partition/bootloader.go

```diff
@@ -127,10 +127,15 @@ type bootloaderType struct {
 }
 
 func newBootLoader(partition *Partition, bootloaderDir string) *bootloaderType {
-	// FIXME: is this the right thing to do? i.e. what should we do
-	//        on a single partition system?
-	if partition.otherRootPartition() == nil {
-		return nil
+	// no root partition means we are in an all-snap system
+	if partition.rootPartition() == nil || partition.otherRootPartition() == nil {
```

Comment:
Unrelated nitpicky rant: reading this without context, I wish we had better names for this that made it a bit more clear why we have the one, and the other one.